Source/WebCore/platform/graphics/gstreamer/mse/PlaybackPipeline.cpp

```diff
@@ -326,6 +326,12 @@ void PlaybackPipeline::flush(AtomString trackId)
 
     GST_DEBUG_OBJECT(appsrc, "flush: trackId=%s", trackId.string().utf8().data());
 
+    if (GST_STATE(pipeline()) < GST_STATE_PAUSED) {
+        // Flushing at early stage may result in playback failure with 'not-linked' reason.
+        // So, lets give pipeline some time to complete pre-roll before flushing this track.
+        gst_element_get_state(pipeline(), nullptr, nullptr, 100 * GST_MSECOND);
+    }
+
```

Comment:
Unfortunately I am aware of there being bugs in GStreamer where playbin and/or other autoplugging elements will get confused when a FLUSH occurs before pre-roll. This is technically not the best solution, as it doesn't protect against the following cases:

* When the pre-roll would have happened eventually, but given thread luck 100ms wasn't enough for it to happen.
* When there are two tracks and one of them had not received yet enough data for pre-roll.
* When the pad being flushed has sent enough buffers for pre-roll, but due to the asynchronous operation of many decoder elements (including avdec_h264), the decoder won't send decoded frames downstream until a new buffer arrives, which will trigger checking how many frames have been decoded by decoder threads.

Still, lacking a good solution, and being in an urgency, this delay may be the simplest acceptable solution we can get by for this problem.